src/chainstate/stacks/db/blocks.rs

```diff
@@ -1478,8 +1603,53 @@ impl StacksChainState {
     }
 
     /// Do we have a particular microblock stream given it _indexed head microblock hash_?
-    pub fn has_confirmed_microblocks_indexed(&mut self, index_block_hash: &BlockHeaderHash) -> Result<bool, Error> {
-        StacksChainState::has_block_indexed(&self.blocks_path, index_block_hash)
+    pub fn has_confirmed_microblocks_indexed(&mut self, index_microblock_hash: &BlockHeaderHash) -> Result<bool, Error> {
+        StacksChainState::has_block_indexed(&self.blocks_path, index_microblock_hash)
+    }
+
+    /// How many microblocks are in a given stream?
+    pub fn get_microblock_stream_length(&self, index_anchor_block_hash: &BlockHeaderHash) -> Result<u64, Error> {
+        let sql = "SELECT COUNT(microblock_hash) FROM staging_microblocks WHERE index_block_hash = ?1".to_string();
+        let args = [&index_anchor_block_hash as &dyn ToSql];
+        let cnt = query_count(&self.blocks_db, &sql, &args).map_err(Error::DBError)?;
+        Ok(cnt as u64)
+    }
+
+    /// Given an index anchor block hash, get the index microblock hash for a confirmed microblock stream.
+    pub fn get_confirmed_microblock_index_hash(&mut self, index_anchor_block_hash: &BlockHeaderHash) -> Result<Option<BlockHeaderHash>, Error> {
+        let sql = "SELECT microblock_hash,burn_header_hash FROM staging_microblocks WHERE index_block_hash = ?1 AND sequence = 0 AND processed = 1 LIMIT 1".to_string();
```

Comment:
This sql variable doesn't need to be a String -- the rusqlite functions are all happy to accept a `&'static str`

Reply:
Fixed in 48a05b2c0